src/ledger/fuzz.zig

```diff
@@ -0,0 +1,330 @@
+const std = @import("std");
+const sig = @import("../sig.zig");
+
+const ColumnFamily = sig.ledger.database.ColumnFamily;
+const AtomicU64 = std.atomic.Value(u64);
+
+var total_action_count: AtomicU64 = AtomicU64.init(0);
+
+const allocator = std.heap.c_allocator;
+
+const Data = struct {
+    value: []const u8,
+};
+
+const cf1 = ColumnFamily{
+    .name = "data",
+    .Key = u64,
+    .Value = Data,
+};
+pub const BlockstoreDB = switch (build_options.blockstore_db) {
+    .rocksdb => ledger.database.RocksDB(&.{cf1}),
+    .hashmap => ledger.database.SharedHashMapDB(&.{cf1}),
+};
+
+const build_options = @import("build-options");
+const ledger = @import("lib.zig");
+
+pub fn run(seed: u64, args: *std.process.ArgIterator) !void {
+    const maybe_max_actions_string = args.next();
+    const maybe_max_actions = blk: {
+        if (maybe_max_actions_string) |max_actions_str| {
+            break :blk try std.fmt.parseInt(usize, max_actions_str, 10);
+        } else {
+            break :blk null;
+        }
+    };
+    defer {
+        _ = total_action_count.fetchAdd(1, .monotonic);
+    }
+
+    // NOTE: change to trace for full logs
+    var std_logger = sig.trace.DirectPrintLogger.init(
+        allocator,
+        .debug,
+    );
+    const logger = std_logger.logger();
+
+    var prng = std.rand.DefaultPrng.init(seed);
+    const random = prng.random();
+
+    const rocksdb_path =
+        try std.fmt.allocPrint(allocator, "{s}/ledger/rocksdb", .{sig.FUZZ_DATA_DIR});
+
+    // ensure we start with a clean slate.
+    if (std.fs.cwd().access(rocksdb_path, .{})) |_| {
+        try std.fs.cwd().deleteTree(rocksdb_path);
+    } else |_| {}
+    try std.fs.cwd().makePath(rocksdb_path);
+
+    var db: BlockstoreDB = try BlockstoreDB.open(
+        allocator,
+        logger,
+        rocksdb_path,
+    );
+
+    defer db.deinit();
```

Comment:
```suggestion
    defer db.deinit();
```

Reply:
Can't find the exact commit but this has also been processed. The extra newline was removed.

Reply:
its still there 
```
    var db = try createBlockstoreDB();

    defer db.deinit();
```